### PR TITLE
Fixed unexpected failure when adding an interface component to a device from the Device detail view

### LINKED ISSUE #6537: Adding an Interface component to a Device fails
### Environment
* Nautobot version (Docker tag too if applicable): `next` branch - as of 2024-Nov-13
* Python version:
* Database platform, version:
* Middleware(s):

### Steps to Reproduce
1. Create a Device instance in UI
2. Attempt to add Interface component through Device detail view (GUI)
3.

### Observed Behavior

Error is returned during validation:
```
Virtual Device Contexts
Ethernet1: "" is not a valid UUID.
```

This might potentially be related, but as a part of testing tried attaching an Interface to a VDC and this resulted in a different error message:
```
ValidationError at /dcim/interfaces/add/
{'virtual_device_contexts': ["Virtual Device Context with names ['test'] must all belong to the same device as the interface's device."]}
```

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes #6537 
# What's Changed

Fixed unexpected failure when adding an interface component to a device from the Device detail view.
More specifically fixed `virtual_device_contexts` field on `InterfaceCreateForm` to `DynamicMultipleModelChoiceField` from the incorrect `DynamicModelChoiceField`.
# Screenshots

Screenshot 2024-11-18 at 4 16 11 PM

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by glennmatthews** [comment]
Does this address both of the symptoms described in #6537 or only the first one?

**Comment by HanlinMiao** [comment]
> Does this address both of the symptoms described in #6537 or only the first one?

The first one is fixed. The second is working as intended but I think it's incomplete. When the `device` on the virtual device context is changed, we do not check if the changed device is the same as that of the assigned interfaces. Im going to try and add that.

**Comment by HanlinMiao** [comment]
Screenshot 2024-11-19 at 11 36 57 AM

File: nautobot/dcim/tests/test_api.py [hunk 1]
```diff
@@ -3488,6 +3488,21 @@ def test_patching_primary_ip_success(self):
             vdc.refresh_from_db()
             self.assertEqual(vdc.primary_ip6, ip_v6)
 
+    def test_changing_device_on_vdc_raise_validation_error(self):
```
Comment (glennmatthews) [inline nautobot/dcim/tests/test_api.py:3491]: Similar test needed in `test_views.py`?

Also, should we add a check in the model `clean()` itself?

Comment (HanlinMiao) [reply nautobot/dcim/tests/test_api.py:3491]: Hmmm, how do I compare the previous `device` to the new `device` assigned in model `clean()`, any hints?

Comment (HanlinMiao) [reply nautobot/dcim/tests/test_api.py:3491]: Also, I am not sure if a test is needed in test_views.py. I am just disabling the `device` field and marking it as `non-required`. No validation error should be raised.

Comment (glennmatthews) [reply nautobot/dcim/tests/test_api.py:3491]: https://github.com/nautobot/nautobot/blob/develop/nautobot/ipam/models.py#L1142-L1146

File: nautobot/core/testing/views.py [hunk 2]
```diff
@@ -470,6 +470,7 @@ def test_edit_object_with_permission(self):
                 "path": self._get_url("edit", instance),
                 "data": utils.post_data(update_data),
             }
+            print(self.client.post(**request).content)
```
Comment (glennmatthews) [inline nautobot/core/testing/views.py:473]: remove :)

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**